Miracast/MiracastService/MiracastController.cpp

```diff
@@ -0,0 +1,1968 @@
+                }
+                else if ( gw_ip_addr.empty() && (std::regex_search(popen_buffer, match, goipRegex2)))
+                {
+                    gw_ip_addr = match[1];
+                    MIRACASTLOG_VERBOSE("GO IP addr obtained is %s\n", gw_ip_addr.c_str());
+                }
+            }
+            MIRACASTLOG_VERBOSE("udhcpc output done\n");
+            pclose(popen_file_ptr);
+            popen_file_ptr = nullptr;
+
+            free(current_line_buffer);
+            current_line_buffer = nullptr;
+
+            if (!local_addr.empty()){
+                MIRACASTLOG_VERBOSE("%s is success\n", command);
+                default_gw_ip_addr = gw_ip_addr;
+                break;
+            }
+        }
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return local_addr;
+}
+
+MiracastError MiracastController::start_DHCPServer(std::string interface)
+{
+    MIRACASTLOG_TRACE("Entering...");
+    std::string command = "";
+
+    command = "ifconfig ";
+    command.append(interface.c_str());
+    command.append(" 192.168.59.1 netmask 255.255.255.0 up");
+
+    MIRACASTLOG_VERBOSE("command : [%s]", command.c_str());
+    system(command.c_str());
+#if defined(UDHCPD_BASED_DHCP_SERVER_ENABLED)
+    command = "ps -ax | grep \"udhcpd\" | grep \"miracast\" | awk '{print $1}' | xargs kill -9";
+
+    system(command.c_str());
+    MIRACASTLOG_VERBOSE("command : [%s]\n",command.c_str());
+
+    system("cp /opt/miracast/udhcpd.conf /opt/secure/wifi/udhcpd_miracast.conf -f");
+
+    command = "sed -i 's/NEW_P2P_GRP_IFACE/";
+    command.append(interface.c_str());
+    command.append("/g' /opt/secure/wifi/udhcpd_miracast.conf");
+
+    system(command.c_str());
+    MIRACASTLOG_VERBOSE("command : [%s]", command.c_str());
+
+    command = "udhcpd -S ";
+    command.append("/opt/secure/wifi/udhcpd_miracast.conf");
+#elif defined(DNSMASQ_BASED_DHCP_SERVER_ENABLED)
+    std::ifstream custom_dnsmasq_commands("/opt/miracast_custom_dnsmasq");
+    std::string mcast_dnsmasq;
+    if (custom_dnsmasq_commands.is_open())
+    {
+        std::getline(custom_dnsmasq_commands, mcast_dnsmasq);
+        MIRACASTLOG_INFO("dnsmasq reading from file [/opt/miracast_custom_dnsmasq] as [ %s] ", mcast_dnsmasq.c_str());
+        custom_dnsmasq_commands.close();
+        command = mcast_dnsmasq;
+        command.append(" -i ");
+        command.append(interface.c_str());
+    }
+    else
+    {
+        command = "ps -ax | awk '/dnsmasq -p0 -i/ && !/grep/ {print $1}' | xargs kill -9";
+
+        MIRACASTLOG_VERBOSE("command : [%s]", command.c_str());
+        system(command.c_str());
+
+        command = "/usr/bin/dnsmasq -p0 -i ";
+        command.append(interface.c_str());
+        command.append(" -F 192.168.59.50,192.168.59.230,255.255.255.0,24h --log-queries=extra");
+    }
+#endif
+    MIRACASTLOG_VERBOSE("command : [%s]", command.c_str());
+    system(command.c_str());
+
+#if 0
```

Comment:
do we need this code?

Reply:
It is not required. I removed it as part of https://github.com/rdkcentral/rdkservices/commit/019f3d11389a6271609f0a3153b3a3984f89ca64